internal/resources/fetching/fetchers/gcp/assets_fetcher.go

```diff
@@ -94,33 +79,31 @@ func NewGcpAssetsFetcher(_ context.Context, log *clog.Logger, ch chan fetching.R
 }
 
 func (f *GcpAssetsFetcher) Fetch(ctx context.Context, cycleMetadata cycle.Metadata) error {
-	f.log.Info("Starting GcpAssetsFetcher.Fetch")
-
-	for typeName, assetTypes := range GcpAssetTypes {
-		assets, err := f.provider.ListAllAssetTypesByName(ctx, assetTypes)
-		if err != nil {
-			f.log.Errorf("Failed to list assets for type %s: %s", typeName, err.Error())
-			continue
-		}
-
-		for _, asset := range assets {
-			select {
-			case <-ctx.Done():
-				f.log.Infof("GcpAssetsFetcher.Fetch context err: %s", ctx.Err().Error())
+	f.log.Info("GcpAssetsFetcher.Fetch start")
+	defer f.log.Info("GcpAssetsFetcher.Fetch done")
+	defer f.provider.Clear()
```

Comment:
`Clear()` will clear out the cache used for cloud account metadata, which is accessed by all assets fetched (mostly all hits)

it used to not get cleared at all (see https://github.com/elastic/cloudbeat/issues/2182)
now it gets cleared when each fetcher exits, which is better but also not good, as it should get called when all fetchers are done. i think a better place would be if the registry would be the one calling a `Clear` method on the `Fetcher` interface, allowing them to clear up caches when fetching cycle is done.